Sources/DateScrubber.swift

```diff
@@ -91,9 +131,24 @@ public class DateScrubber: UIViewController {
         self.setFrame(atYpos: yPos)
     }
 
-    public func updateSectionTitle(title : String){
-        self.sectionLabel.setText(title)
-        self.setSectionlabelFrame()
+    public func updateSectionTitle(title: String) {
+
+        if self.currentSectionTitle != title {
+            self.currentSectionTitle = title
+
+            self.sectionLabel.setText(title)
+            self.setSectionlabelFrame()
+        }
+    }
+
+    private func userInteractionOnScrollViewDetected(){
+
+        NSObject.cancelPreviousPerformRequestsWithTarget(self, selector: #selector(hideScrubber), object: nil)
```

Comment:
I didn't understood much of the bug, but you can cancel `performSelector...` using 

```swift
NSObject.cancelPreviousPerformRequestsWithTarget(self)
```

Maybe that helps